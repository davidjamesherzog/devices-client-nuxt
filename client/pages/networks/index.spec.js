import { createLocalVue, shallowMount } from '@vue/test-utils';
import Buefy from 'buefy';
import Networks from './index.vue';

describe('Networks', () => {
  describe('fetch', () => {
    test('should return networks', async (done) => {
      const networkData = [
        {
          id: 1,
          name: 'GozrehblasterN',
          description: 'Test Wifi',
          type: 'WIFI'
        },
        {
          id: 2,
          name: 'HerzogHomeN',
          description: 'Test Wifi',
          type: 'WIFI'
        },
        {
          id: 3,
          name: 'Wired',
          description: 'Wired devices',
          type: 'Wired'
        }
      ];
      const networks = new Networks();
      networks.$axios = {
        $get: jest.fn().mockReturnValue(networkData)
      };
      const data = await networks.fetch();
      expect(networks.$axios.$get).toHaveBeenCalledWith('/networks');
      expect(data).toEqual(networkData);
      done();
    });
  });

  describe('template', () => {
    let localVue;

    beforeEach(() => {
      localVue = createLocalVue();
      localVue.use(Buefy);
    });

    test('is a Vue instance', () => {
      const wrapper = shallowMount(Networks, { localVue });
      expect(wrapper.vm).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
