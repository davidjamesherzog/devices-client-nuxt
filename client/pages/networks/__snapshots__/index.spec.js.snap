// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Networks template is a Vue instance 1`] = `
<div>
  <section class="section">
    <h2 class="title is-3 has-text-grey">Networks</h2>
  </section>
  <section class="networks"></section>
</div>
`;
